apps/hash-graph/lib/graph/Cargo.toml

```diff
@@ -11,12 +11,15 @@ type-fetcher = { path = "../type-fetcher", optional = true }
 async-trait = "0.1.66"
 axum = "0.6.11"
 bb8-postgres = "0.8.1"
+bytes = "1.4.0"
 clap = { version = "4.1.9", features = ["derive", "env"], optional = true }
 time = { version = "0.3.20", features = ['serde', 'serde-well-known'] }
 derivative = "2.2.0"
 error-stack = { version = "0.3.1", features = ["spantrace"] }
 futures = "0.3.27"
+hash-status = { path = "../../../../libs/@local/status/crate" }
```

Comment:
We're going to try and rely on Turbo's dependency graph to decide what CI checks to run for Node things. I think we should commit to that as a repo-level solution and rely on it for Rust as well so we can remove our bespoke logic

Reply:
Yes, this is possible, but this will greatly slow down CI times. Given this example, the graph depends on `status`, but also, let's say, 5 other packages depend on `status`, all 6 packages need to be tested. While Turbo is able to test all of those packages, every test would run in the same GitHub Runner. Unless it's possible to let Turbo generate a `matrix` object (like the current `setup,py`), all of our test CI jobs would run in a single, gigantic job and it's impossible to re-run only a single, failed job. Especially for the graph (which has currently one of the longest-running CIs for testing), this implies compiling it for different configurations with different targets multiple times (e.g. debug, release, test, binaries, integration, docker), which are currently in various jobs, which run parallel on different runners. 

If we can use turbo to create a list of packages, which needs to be tested, we can use that as our entry point for the CI and I'm happy to do some discovery here, so this could almost entirely replace the current `setup.py` job.